en/concept/architecture.md

```diff
@@ -1,36 +1,121 @@
-  * Communication is done by asynchronous message passing 
-  * The system can deal with varying workload
+- All functionality is divided into exchangeable components
+- Communication is done by asynchronous message passing 
+- The system can deal with varying workload
 
 In addition to these runtime considerations, its modularity maximizes [reusability](https://en.wikipedia.org/wiki/Reusability).
 
+
 ## High Level Software Architecture
 
-Each of the blocks below is a separate module, which is self-contained in terms of code, dependencies and even at runtime. Each arrow is a connection through publish/subscribe calls through [uORB](../middleware/uorb.md).
+The diagram below provides an overview of the building blocks of PX4. 
+The top part of the diagram contains middleware blocks, while the lower
+section shows the components of the flight stack.
+
+![PX4 Architecture](../../assets/diagrams/PX4_Architecture.svg)
+
+<!-- This diagram can be updated from 
+[here](https://drive.google.com/file/d/0B1TDW9ajamYkaGx3R0xGb1NaeU0/view?usp=sharing) 
+and opened with draw.io Diagrams. -->
+
+The source code is split into self-contained modules (shown in `monospace` in the
+diagram). Usually a building block corresponds to exactly one module. At
+runtime, you can inspect which modules are executed with the `top` command, and
+each module can be started/stopped individually via `<module_name> start/stop`.
+
+The arrows show the information flow for the *most important* connections between
+the modules. In reality, there are many more connections than shown, and some messages 
+(e.g. for parameters) are accessed by most of the modules.
+
+Modules communicate with each other through a publish-subscribe message bus
+named [uORB](../middleware/uorb.md). 
+The use of the publish-subscribe scheme means that:
+
+- The system is reactive — it will update instantly when new data is available
+- All operations and communication are fully parallelized
+- A system component can consume data from anywhere in a thread-safe fashion
+
+> **Info** This architecture allows every single one of these
+> blocks to be rapidly and easily replaced, even at runtime.
+
+The controllers / mixers are specific to a particular airframe (e.g. a
+multicopter, VTOL or plane), but the higher-level mission management blocks like
+the `commander` and `navigator` are shared between platforms.
+
+### Middleware {#middleware}
+
+The middleware consists primarily of device drivers
+for embedded sensors, communication with the external world (companion computer,
+GCS, etc.) and a publish-subscribe message bus.
+
+### Flight Stack {#flight-stack}
+
+The flight stack is a collection of guidance, navigation and control algorithms 
+for autonomous drones. 
+It includes controllers for fixed wing, multirotor and VTOL airframes 
+as well as estimators for attitude and position.
+
+## Update Rates
+
+Since the modules poll for message updates, typically the drivers define how
```

Comment:
The fact that modules poll for message updates is news to me (and the reader) here. It is also a bit confusing since above we said the system was reactive, so I would have assumed that a subscriber would immediately get notification.

Perhaps best to start this with ...
`Modules poll for message updates at a certain rate. `